toolkit.lisp

```diff
@@ -66,93 +66,4 @@
-          (incf (aref histogram (floor (* (random 1.0 rng) bins)))
-                sample-contribution))))
-    (format stream " 100%~%")
-    (let ((duration (/ (- (get-internal-real-time) start)
-                       INTERNAL-TIME-UNITS-PER-SECOND)))
-      (format stream "Generation took: ~6,3fs, ~fμs/sample~%"
-              duration (* 1000000 (/ duration samples))))
-    histogram))
-
-(defun print-histogram (histogram &key (stream *standard-output*) (width 80))
-  (assert (< 7 width))
-  (let ((half-width (/ (- width 7) 2))
-        (total 0.0))
-    (loop for bin across histogram
-          for deviation = (* 100 (- bin (/ (length histogram))))
-          for chars = (max (- half-width) (min (+ half-width) (floor (* 2 half-width deviation))))
-          do (format stream "~6,3@f% ~v@{░~}~v@{█~}~v@{░~}~%" deviation
-                     (min half-width (+ half-width chars))
-                     (abs chars)
-                     (min half-width (- half-width chars))
-                     NIL)
-             (incf total (abs deviation)))
-    (format stream "Cumulative deviation: ~6,3f%~%" total)
-    histogram))
-
-(defun benchmark (rng &key (samples 1000000) (stream *standard-output*))
-  (let* ((rng (ensure-generator rng))
-         (next-fun (next-byte-fun rng))
-         (start (get-internal-run-time)))
-    (declare (type (unsigned-byte 64) samples))
-    (declare (type (function (generator) T) next-fun))
-    (locally (declare (optimize speed (safety 0)))
-      (loop repeat samples
-            do (funcall next-fun rng)))
-    (let* ((end (get-internal-run-time))
-           (duration (float (/ (- end start) INTERNAL-TIME-UNITS-PER-SECOND) 0d0)))
-      (format stream "Duration: ~12t~10,1f~%Samples: ~12t~10d~%Samples/s: ~12t~10,1f~%S/sample: ~12t~10,8f"
-              duration samples (/ samples duration) (/ duration samples))
-      (/ samples duration))))
-
-(defun benchmark-all (&key (samples 1000000) (stream *standard-output*))
-  (let* ((nullstream (make-broadcast-stream))
-         (stats (loop for rng in (list-generator-types)
-                      for result = (cons rng (handler-case (benchmark rng :samples samples :stream nullstream)
-                                               (error () -1)))
-                      when result collect result)))
-    (setf stats (sort stats #'> :key #'cdr))
-    (format stream "RNG~20tSamples/second~%")
-    (loop for (rng . samples) in stats
-          do (format stream "~&~a~20t~18,1f~%" rng samples))))
-
-(defun list-dim (list)
-  (list* (length list)
-         (when (listp (first list))
-           (list-dim (first list)))))
-
-(defmacro %arr (type &rest elements)
-  `(make-array ',(list-dim elements)
-               :element-type ',type
-               :initial-contents ',elements))
-
-(defmacro define-pregenerated (name contents)
-  `(progn
-     (let ((contents ,contents))
-       (defun ,name () contents))
-     (define-symbol-macro ,name (,name))))
```

Comment:
why are all these functions removed?

Reply:
That is a very good question. My best guess is that I did it inadvertently when I rebased this change on top of your latest master. I will go back and fix it.

Reply:
I think this is fixed now.